app/assets/javascripts/modules/highlight-active-section-heading.js

```diff
@@ -0,0 +1,136 @@
+;(function (Modules, root) {
+  'use strict'
+
+  var $ = root.$
+  var $window = $(root)
+
+  Modules.HighlightActiveSectionHeading = function () {
+    var self = this
+    var _hasResized = true
+    var _hasScrolled = true
+    var _interval = 50
+    var anchorIDs = []
+
+    self.getWindowDimensions = function () {
+      return {
+        height: $window.height(),
+        width: $window.width()
+      }
+    }
+
+    self.getWindowPositions = function () {
+      return {
+        scrollTop: $window.scrollTop()
+      }
+    }
+
+    self.getElementOffset = function ($el) {
+      return $el.offset()
+    }
+
+    self.start = function ($el) {
+      $window.resize(self.hasResized)
+      $window.scroll(self.hasScrolled)
+
+      setInterval(self.checkResize, _interval)
+      setInterval(self.checkScroll, _interval)
+
+      self.$anchors = $el.find('.js-page-contents a')
+      self.totalAnchors = self.$anchors.length
+      self.getAnchors(self.totalAnchors)
+
+      self.checkResize()
+      self.checkScroll(self.totalAnchors)
+    }
+
+    self.hasResized = function () {
+      _hasResized = true
+      return _hasResized
+    }
+
+    self.hasScrolled = function () {
+      _hasScrolled = true
+      return _hasScrolled
+    }
+
+    self.checkResize = function () {
+      if (_hasResized) {
+        _hasResized = false
+        _hasScrolled = true
+      }
+    }
+
+    self.checkScroll = function () {
+      if (_hasScrolled) {
+        _hasScrolled = false
+        var windowDimensions = self.getWindowDimensions
+        if (windowDimensions <= 768) {
+          self.removeActiveItem()
+        } else {
+          self.updateActiveNavItem()
+        }
+      }
+    }
+
+    self.getAnchors = function (totalAnchors) {
+      for (var i = 0; i < self.totalAnchors; i++) {
+        var anchor = self.$anchors[i]
+        var anchorID = $(anchor).attr('href')
```

Comment:
If $anchors is an array of jQuery objects, do you need to call jQuery again? `anchor.attr`should work? If so, worth naming `anchor` `$anchor` instead?